### PR TITLE
Add CONTRIBUTING.md to the docs

### PR DESCRIPTION
**Pull request by jtpavlock**
## Description

This does a few things:
1. Converts CONTRIBUTING.md to CONTRIBUTING.rst - this allows it to be in the docs
2. Adds CONTRIBUTING.rst to the docs
3. Adds the Testing wiki to CONTRIBUTING.rst
4. Adds a table of contents to the top of CONTRIBUTING.rst

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+   “bite-sized” <https://github.com/beetbox/beets/labels/bitesize>`__.
+   These are issues that would serve as a good introduction to the
+   codebase. Claim one and start exploring!
+-  Like testing? Our `test
+   coverage <https://codecov.io/github/beetbox/beets>`__ is somewhat
+   low. You can help out by finding low-coverage modules or checking out
+   other `testing-related
+   issues <https://github.com/beetbox/beets/labels/testing>`__.
+-  There are several ways to improve the tests in general (see
+   `Testing <https://github.com/beetbox/beets/wiki/Testing>`__) and some
```
Comment (sampsyo) [inline CONTRIBUTING.rst:102]: Maybe change this to a section reference?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+   contribute tests that show that your code does what it says.
+4. Add documentation. If you’ve added a new command flag, for example,
+   find the appropriate page under ``docs/`` where it needs to be
+   listed.
+5. Add a changelog entry to ``docs/changelog.rst`` near the top of the
+   document.
+6. Run the tests and style checker. The easiest way to run the tests is
+   to use `tox <https://tox.readthedocs.org/en/latest/>`__. For more
+   information on running tests, see our `Testing wiki
+   page <https://github.com/beetbox/beets/wiki/Testing>`__.
```
Comment (sampsyo) [inline CONTRIBUTING.rst:153]: As above.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+   -  ``self.assertEqual``
+   -  ``self.assertTrue``
+   -  ``self.assertFalse``
+   -  ``self.assertRaises``
+
+-  For detailed information see `Python unittest`_
+-  **AVOID** using the ``start()`` and ``stop()`` methods of
+   ``mock.patch``, as they require manual cleanup. Use the annotation or
+   context mana
```
Comment (sampsyo) [inline CONTRIBUTING.rst:345]: Looks like this got cut off somehow?

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+as a test runner, we prefer to write tests using the standard
+`unittest`_ testing framework.
+
+Any tests that involve sending out network traffic e.g. an external API
+call, should be skipped normally and run under our weekly `integration
+test`_ suite. These tests can be useful in detecting external changes
+that would affect ``beets``. In order to do this, simply add the
+following snippet before the applicable test case:
+
+::
```
Comment (sampsyo) [inline CONTRIBUTING.rst:319]: Shorter way to do this in ReST:

```suggestion
following snippet before the applicable test case::
```

**Comment by jtpavlock** [comment]
Thanks for the sanity check. I fixed the issues you mentioned.

> Over time, it might be nice to split this stuff up into multiple pages… there's a lot of information here, and it covers a pretty wide range from "consider writing a blog post!" to "here are the functions to use when printing a path string." We could make the "main page" a simple overview and keep the nitty-gritty details in the docs only, for example.

True, I'm hoping the table of contents I added at the top will also help with navigating around it. Either way, I think this is a good first step at getting it in the docs.

**Comment by sampsyo** [comment]
Indeed. In any case, I think this is good to go and will hit the green button—possible reorganizations can happen "off the critical path." :smiley:

**Pull request merged**